config/config/src/main/java/io/helidon/config/ConfigMapperManager.java

```diff
@@ -161,6 +161,7 @@ static final class MapperProviders {
                 Optional<? extends BiFunction<Config, ConfigMapper, ?>>>> providers = new LinkedList<>();
 
         private MapperProviders() {
+            add(new EnumMapperProvider());
```

Comment:
This struck me as odd; is there something that makes this more primordial than other providers? (I am coming to this without a deep knowledge of the config system so there very well may be such a thing.)

Reply:
Yea, basically I am suggesting the same thing in one of my comments as well.

Reply:
I think it might strike you as odd only because this is the first built-in conversion that needs to be a mapper _provider_ and not simply a mapper.

All the other built-in conversions can be (and are) implemented as a single mapper for each type conversion, Concretely, we need only a single mapper for `Integer`, for example, because all integers obey the same conversion rules. 

Not so for `enum`s because each `enum` type needs its own mapper that knows which values are legal for _that_ `enum`.

Those other converters--being mappers--are plugged into the config infrastructure in one way. Being a mapper provider, this new one get plugged in differently.

Reply:
Ah, makes sense. Feel free to resolve the comment.

Reply:
Actually it is analogous to Integer.
		Mytype.valueOf(<configValStr>)

Reply:
Not at all analogous in the very general config mapping world where this is all happening.

Yes, both `Integer` and `MyEnum` have a `valueOf` method. But that's not enough to allow us to handle them the same way in value mapping.